charts/kueue/templates/certmanager/cacertificate.yaml

```diff
@@ -0,0 +1,26 @@
+# Add CA certificate and issuer
+{{- if .Values.enableCertManager }}
```

Comment:
I was expecting just new tests in this PR.

Does it indicate this is missing to support external certs for metrics in 0.11? If so we may need to handle this as a feature / bugfix, not just tests. cc @kannon92

Reply:
Yeah, that's the case. I can split the existing commit into two: one for fixing the metrics certs and the second for adding tests.

Reply:
I see , I'm ok to keep tests together with the fix actually, just wasn't sure about the status. @kannon92 can you review the fixing part?

Reply:
I didn't need this with manually testing prometheus. But it seems to be needed when testing curl with metrics

Reply:
I was trying to check again on metrics certs and I am having trouble verifying that this works..

E2E test would be nice!

I'll investigate more tomorrow.

/hold

Reply:
IMO this shouldn't be necessary. A simple self-signed certificate works for me, both with prometheus and with curl. This is the pod I tested this with: https://gist.github.com/atiratree/6b8d0d6b5c63b546f86d542a6080cd59

Btw, I am enabling TLS metrics in the jobset, so you can also take an inspiration from https://github.com/kubernetes-sigs/jobset/pull/863  in case something is misconfigured in kueue.

Reply:
I think I figured out my issue.

https://github.com/kubernetes-sigs/kueue/pull/4903